application/api/controller.py

```diff
@@ -0,0 +1,462 @@
+                        return make_response(
+                            jsonify(
+                                {
+                                    'message': "Bucketlist created successfully.",
+                                    "name": name,
+                                    "description": description,
+                                    "created_by": created_by
+                                }
+                            ), 201)
+                else:
+                    return make_response(
+                        jsonify(
+                            {
+                                'message': user_id
+                            }
+                        ), 401)
+        except:
+            return make_response(jsonify({
+                "error": "missing data in request."
+            }))
+            
+    def get(self):
+        token = request.headers.get('Authorization')
+        user_id = User.verify_token(token)
+        if isinstance(user_id, int):
+            user_bucketlists = BucketList.query.filter_by(
+                created_by=user_id).all()
+            bucketlists = []
+            if user_bucketlists:
+                for b_lists in user_bucketlists:
+                    bucketlists.append({
+                        "id": b_lists.id,
+                        "name": b_lists.name,
+                        "description": b_lists.description,
+                        "created_by": b_lists.created_by
+                    })
+                return make_response(jsonify({'bucketlists': bucketlists}), 201)
+
+            else:
+                return make_response(
+                    jsonify(
+                        {
+                            'message': "There are no bucketlists for the current user."
+
+                        }
+                    ), 404)
+        else:
+            return make_response(
+                jsonify(
+                    {
+                        'message': user_id
+
+                    }
+                ), 401)
+
+
+class SingleBucketList(Resource):
+    """Shows a single bucketlist item and lets you delete a bucketlist item."""
+
+    def put(self, bucketlist_id):
+
+        token = request.headers.get('Authorization')
+        user_id = User.verify_token(token)
+        if isinstance(user_id, int):
+            bucketlist = BucketList.query.filter_by(
+                created_by=user_id).first()
+            if not bucketlist:
+                return make_response(
+                    jsonify(
+                        {
+                            'message': "Bucketlist not found."
+
+                        }
+                    ), 404)
+
+            new_bucketlist_name = request.json["name"]
+            if not new_bucketlist_name:
+                return make_response(
+                    jsonify(
+                        {
+                            'message': "name field is missing."
+
+                        }
+                    ), 403)
```

Comment:
status code `400` would be more appropriate here considering `403` refers to `The server understood the request but refuses to authorize it.`. In this case the server cannot process the request with the missing parameters.